pkg/platform/packet/packet.go

```diff
@@ -33,21 +35,31 @@ import (
 	"github.com/kinvolk/lokomotive/pkg/terraform"
 )
 
+type nodeRole int
+
+const (
+	controller = iota
+	worker
+)
+
 type workerPool struct {
-	Name           string `hcl:"pool_name,label"`
-	Count          int    `hcl:"count"`
-	DisableBGP     bool   `hcl:"disable_bgp,optional"`
-	IPXEScriptURL  string `hcl:"ipxe_script_url,optional"`
-	OSArch         string `hcl:"os_arch,optional"`
-	OSChannel      string `hcl:"os_channel,optional"`
-	OSVersion      string `hcl:"os_version,optional"`
-	NodeType       string `hcl:"node_type,optional"`
-	Labels         string `hcl:"labels,optional"`
-	Taints         string `hcl:"taints,optional"`
-	SetupRaid      bool   `hcl:"setup_raid,optional"`
-	SetupRaidHDD   bool   `hcl:"setup_raid_hdd,optional"`
-	SetupRaidSSD   bool   `hcl:"setup_raid_ssd,optional"`
-	SetupRaidSSDFS bool   `hcl:"setup_raid_ssd_fs,optional"`
+	Name                  string            `hcl:"pool_name,label"`
+	Count                 int               `hcl:"count"`
+	DisableBGP            bool              `hcl:"disable_bgp,optional"`
+	IPXEScriptURL         string            `hcl:"ipxe_script_url,optional"`
+	OSArch                string            `hcl:"os_arch,optional"`
+	OSChannel             string            `hcl:"os_channel,optional"`
+	OSVersion             string            `hcl:"os_version,optional"`
+	NodeType              string            `hcl:"node_type,optional"`
+	Labels                string            `hcl:"labels,optional"`
+	Taints                string            `hcl:"taints,optional"`
+	ReservationIDs        map[string]string `hcl:"reservation_ids,optional"`
+	ReservationIDsDefault string            `hcl:"reservation_ids_default,optional"`
+	SetupRaid             bool              `hcl:"setup_raid,optional"`
+	SetupRaidHDD          bool              `hcl:"setup_raid_hdd,optional"`
+	SetupRaidSSD          bool              `hcl:"setup_raid_ssd,optional"`
+	SetupRaidSSDFS        bool              `hcl:"setup_raid_ssd_fs,optional"`
+	NodesDependsOn        string            // Not exposed to the user
```

Comment:
I would make it `[]string` and then render it nicely in the template.

Reply:
Good point, thanks!

I was in the fence on that, using json marshall or this. But I think the current approach is the one I dislike the most, will try json or rendering (json doesn't seem so appealing because it will be in the struct itself, instead in other fields as all [other parts of the code do](https://github.com/kinvolk/lokomotive/blob/master/pkg/platform/packet/packet.go#L194-L196))

Reply:
Done :)

Reply:
Thanks for the effort!